linfa-ensemble/src/random_forest/algorithm.rs

```diff
@@ -0,0 +1,191 @@
+use crate::random_forest::hyperparameters::RandomForestParams;
+use linfa_predictor::{LinfaError, Predictor, ProbabilisticPredictor};
+use linfa_trees::DecisionTree;
+use ndarray::Array;
+use ndarray::Axis;
+use ndarray::{Array1, ArrayBase, Data, Ix1, Ix2};
+use ndarray_rand::rand_distr::Uniform;
+use ndarray_rand::RandomExt;
+use std::collections::HashMap;
+
+/// A random forest is composed of independent decision trees performing a prediction and collecting each of them
```

Comment:
nit: Maybe something to the effect of "whose predictions are aggregated" instead of "collecting"